grimoire/districts/merging_districts.py

```diff
@@ -1,13 +1,13 @@
-from ..districts.district import District
+from ..districts.district import District, SuperDistrict
+from ..districts.district_analyze import district_analyze, get_candidate_score
+from ..core.maps import Map
 
-RURAL_SIZE_RATIO = (
-    3  # we expect rural districts to be this times larger in area than urban ones
-)
+RURAL_SIZE_RATIO = 3  # NOTE: not currently used, we expect rural districts to be this times larger in area than urban ones
 
 
 # Merges districts until we have a target number reached
 def merge_down(
-    districts: list[District], district_map: list[list[District]], target_number: int
+    districts: list[District], district_map: list[list[District]], target_number: int, map: Map
 ):
     identities = {
```

Comment:
@Jandhi You created this originally, if you want to comment more.

Its mostly used during merge to ensure the district identities don't get mixed up. Since we end up creating a lot of districts we end up merging into others, we use this to ensure adjacency and other values remain consitent.